contactsummary/tests/phpunit/CRM/Contactsummary/Utils/AbsencesTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+require_once 'CiviTest/CiviUnitTestCase.php';
+require_once 'ContractSummaryTestTrait.php';
+
+class CRM_Contactsummary_Utils_AbsencesTest extends CiviUnitTestCase {
+
+  use ContractSummaryTestTrait;
+
+  function setUp() {
+    $this->cleanDB();
+    parent::setUp();
+    $jobContractUpgrader = CRM_Hrjobcontract_Upgrader::instance();
+    $jobContractUpgrader->install();
+    $hrAbsenceUpgrader = CRM_HRAbsence_Upgrader::instance();
+    $hrAbsenceUpgrader->installAbsenceTypes();
+    $params = array(
+      'label' => 'Absence',
+      'filter' => 1,
+      'is_active' => 1,
+      'is_optgroup' => 0,
+      'is_default' => 0,
+      'grouping' => 'Timesheet',
+    );
+   civicrm_api3('activity_type', 'create', $params);
+  }
+
+  function tearDown() {
+    parent::tearDown();
+  }
+
+  function testGetTotalAbsencesForSickLeaveType() {
+
+    // create absence periods
+    $StartDate1 = date('Y-m-d H:i:s', strtotime('-1 year'));
+    $EndDate1 = date('Y-m-d H:i:s', strtotime('+1 year'));
+    $StartDate2 = date('Y-m-d H:i:s', strtotime('-4 year'));
+    $EndDate2 = date('Y-m-d H:i:s', strtotime('-3 year'));
+    $StartDate3 = date('Y-m-d H:i:s', strtotime('+3 year'));
+    $EndDate3 = date('Y-m-d H:i:s', strtotime('+4 year'));
```

Comment:
For numbers bigger than 1, it's better to use the plural form. Example: -3 years, +4 years